src/System.Private.ServiceModel/tests/Scenarios/Encoding/Encoders/TextEncodingTests.cs

```diff
@@ -48,4 +48,36 @@ public static void SameBinding_SecurityModeNone_Text_EchoString_Roundtrip()
 
         Assert.True(errorBuilder.Length == 0, "Test case FAILED with errors: " + errorBuilder.ToString());
     }
+
+    [Fact]
+    [OuterLoop]
+    public static void TextMessageEncoder_WrongContentTypeResponse_Throws_ProtocolException()
+    {
```

Comment:
It would still be good to test what happens when you set the content type to some legal parsable value *other* than the normal one for the response.  Does it flow back to the client as is? Cause a different kind of exception on the client?  Cause an exception on the server?